src/services/shims.ts

```diff
@@ -779,6 +798,35 @@ module ts {
                 });
         }
 
+        public getTSConfigFileInfo(fileName: string, sourceTextSnapshot: IScriptSnapshot): string {
+            return this.forwardJSONCall(
+                "getTSConfigFileInfo('" + fileName + "')",
+                () => {
+                    let text = sourceTextSnapshot.getText(0, sourceTextSnapshot.getLength());
+
+                    try {
+                        var json = /\S/.test(text) ? JSON.parse(text) : {};
+                    }
+                    catch (e) {
+                        return {
+                            options: {},
+                            files: [],
+                            errors: realizeDiagnostic(createCompilerDiagnostic(Diagnostics.Unable_to_open_file_0, fileName), '\r\n')

```

Comment:
+1 for showing the error message.

also you should use ts.readConfigFile from commandlineParser.ts. and add the error info. see #2627 for more details. i would give it additional reportError function that would do that, and throw here as well.